### PR TITLE
Persistent cache with RocksDB

### LINKED ISSUE #36: Persistent Cache is available
### Background:
The existing in-memory cache in mapget serves its purpose for short-lived sessions but poses limitations for long-term or recurring data retrieval needs. This transient nature of the cache can lead to repetitive data fetching from source, negatively impacting performance and user experience. Implementing a persistent cache would alleviate this issue by storing data between sessions, saving time and computational resources. 

### Goal:
Introduce a persistent caching mechanism in mapget to improve long-term data retrieval efficiency and to complement the existing in-memory cache.

### Acceptance Criteria:

1. **Persistence:** The cache should be capable of storing data between different sessions and system reboots.
  
2. **Compatibility:** The persistent cache should function alongside the existing in-memory cache, allowing users the flexibility to choose between the two depending on their needs.
  
3. **Size Limit:** A configurable maximum cache size is essential to prevent overutilization of disk space, also support 'unlimited'. 

4. **Data Integrity:** Include mechanisms to ensure data consistency and integrity.

5. **Cache Flushing:** Users should have the ability to manually clear the persistent cache when needed.
  
6. **Future-Proofing:** Like the in-memory cache, the implementation should be designed to be compatible with other caching mechanisms that may be introduced later.

### Additional Thoughts/Ideas:

**Configurable TTL (Time-to-Live):** Users should be able to configure the expiration time for data stored in the persistent cache.

### PR DESCRIPTION
**Pull request by l-laura**
Available for code review, check out the unit tests in `test-cache.cpp` for an overview of the functionality.

Integration tests:
1. Set up datasource with static GeoJSON tiles as described in [the pull request to Polylines](https://github.com/ndsev/erdblick/pull/42).
2. Run `mapget` with cache options, for example:
 ```
mapget serve --cache-type rocksdb --cache-dir "custom-cache-dir" --cache-max-tiles 30 --clear-cache 1
 ```
and jump to the tiles in `erdblick`. Debug logs contain output about cache hits and misses. Log level needs to be set via environment variable, I will investigate why the CLI flag doesn't work in the scope of the next ticket.

Closes #36

### CODE REVIEW

**Review by MisterGC — DISMISSED** [review]
I have not checked the source but only tested this PR from user perspective.

**Environment:**
* Dockerized devenv which now also has a dedicated [launch config for the source with persistent cache](https://github.com/ndsev/mapviewer/commit/dab923e33efe581aa336e959ce1f694bcf9699ad)
* Host system Linux (Ubuntu 22.04 based)
* Browser: Firefox 118.0.2
* in order to being able to have the exact same map locations I have used `ebDebug.set/getCamera`

**Things I have tested:**
* used GeoJSON data source with persistent cache activate
* check that the clearing options works (kept content for two sessions, then cleared it)
* checked that the data gets fetched from the cache by checking loading times visually for a ~500 tiles viewport
* switched to in-memory cache to check if it still works
* I have stopped mapget between sessions by just using `stop debugging` of VS code (just because we have discussed potential cache corruption today)

**What I haven't tested(just assuming it works or isn't essential/critical to move forward):** 
- cache logging in the log file
- running it on other OS with other browser
- varying the values of all other cmd line options beside type and clearing
- long-running sessions

**Result:** The behavior looks good, therefore I'd suggest to merge this one as soon as all build succeed or do I miss any essentials aspect? Also let me know if there is something which should be double-checked in addition.

**Comment by MisterGC** [comment]
I have now further investigated and found some issues:
- sometimes the `LOCK` file stays when I stop the source (with `SIGTERM`), either we avoid this kind of locking behavior or we ensure that it gets properly reset on-demand at ever
- when keeping the cache from one session to another, there seems to be issues with restoring the field cfg - I recognized this by always using the same style and getting different results, the property I have used in the property was no longer available when the layer has been restored from the cache - have a look at the following screenshots, the first one is the intial one, the show how it looks like when using the persistent cache:

style_as_expected
style_partially_applied
style_partially_applied2

File: libs/http-service/src/cli.cpp [hunk 1]
```diff
@@ -18,56 +18,103 @@ struct ServeCommand
...
+            "Data sources in format <host:port>. Can be specified multiple times.");
+        serveCmd->add_option(
+            "-e,--datasource-exe",
+            datasourceExecutables_,
+            "Data source executable paths, including arguments. "
+            "Can be specified multiple times.");
+        serveCmd->add_option(
+            "-c,--cache-type", cacheType_, "From [memory|rocksdb], default memory.")
+            ->default_val("memory");
+        // TODO should RocksDB be the default cache?
```
Comment (josephbirkner) [inline libs/http-service/src/cli.cpp:47]: I vote yes 👍 But we should document the default somehow (also the default max cache size and path).

File: libs/http-service/src/cli.cpp [hunk 2]
```diff
@@ -18,56 +18,103 @@ struct ServeCommand
...
-        HttpService srv;
 
-        if (!datasourceHosts_.empty()){
-            for (auto& ds : datasourceHosts_){
+        std::shared_ptr<Cache> cache;
+        if (cacheType_ == "rocksdb") {
+            cache = std::make_shared<RocksDBCache>(cacheMaxTiles_, cachePath_, clearCache_);
+        }
+        else if (cacheType_ == "memory") {
+            // TODO support setting max size also for in-memory cache.
```
Comment (josephbirkner) [inline libs/http-service/src/cli.cpp:73]: This should indeed be done, just pass `cacheMaxTiles_` in the constructor.

File: test/unit/test-cache.cpp [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        auto feature = tile->newFeature("Way", {{"areaId", "MediocreArea"}, {"wayId", 24}});
+        cache->putTileFeatureLayer(tile);
+        auto updatedTile = cache->getTileFeatureLayer(tile->id(), info);
+        assert(updatedTile->size() == 1);
+
+        // Check that cache hits and misses are properly recorded.
+        auto missingTile = cache->getTileFeatureLayer(otherTile->id(), otherInfo);
+        assert(cache->getStatistics()["cache-hits"] == 2);
+        assert(cache->getStatistics()["cache-misses"] == 1);
+        // TODO discuss: cache hit/miss stats are not persistent, is that a problem?
```
Comment (josephbirkner) [inline test/unit/test-cache.cpp:127]: Not a problem.

**Pull request merged**